src/ftest.jl

```diff
@@ -0,0 +1,130 @@
+type FTestTable
+    ntests::Int
+    SSR1::Array{Float64, 1} # sum of squared residuals, model 1
+    SSR2::Array{Float64, 1} # sum of squared residuals, model 2
+    df1::Array{Int, 1} # degrees of freedom, model 1 residuals
+    df2::Array{Int, 1} # degrees of freedom, model 2 residuals
+    MSR1::Array{Float64, 1} # mean of squared residuals, model 1
+    MSR2::Array{Float64, 1} # mean of squared residuals, model 2
+    fstat::Array{Float64, 1} # f statistic
+    pval::Array{Float64, 1} # p value
+end
+
+function FTestTable(SSR1::Array{Float64, 1}, SSR2::Array{Float64, 1}, df1::Array{Int, 1}, df2::Array{Int, 1}, MSR1::Array{Float64, 1}, MSR2::Array{Float64, 1}, fstat::Array{Float64, 1}, pval::Array{Float64, 1})
+#Note: Is checking args in internal helper functions the desired behavior?
+@argcheck length(SSR1) == length(SSR2) == length(df1) == length(df2) == length(MSR1) == length(MSR2) == length(fstat) == length(pval) "F test values  must all be the same length"
+
+    return FTestTable(length(SSR1), SSR1, SSR2, df1, df2, MSR1, MSR2, fstat, pval)
+end
+
+function FTestTable(SSR1::Float64, SSR2::Float64, df1::Int, df2::Int, MSR1::Float64, MSR2::Float64, fstat::Float64, pval::Float64)
+    return  FTestTable([SSR1], [SSR2], [df1], [df2], [MSR1], [MSR2], [fstat], [pval])
+end
+
+"""A helper function to determine if mod1 is nested in mod2"""
+function issubmodel(mod1::LinPredModel, mod2::LinPredModel)
+    mod1.rr.y != mod2.rr.y && return false # Response variables must be equal
+
+    # Now, test that all predictor variables are equal
+    pred1 = mod1.pp.X
+    npreds1 = size(pred1, 2)
+    pred2 = mod2.pp.X
+    npreds2 = size(pred1, 2)
+    # If model 1 has more predictors, it can't possibly be a submodel
+    npreds1 > npreds2 && return false 
+    
+    @inbounds for i in 1:npreds1
+        var_in_mod2 = false
+        for j in 1:npreds2
+            if view(pred1, :, i) == view(pred2, :, j)
+                var_in_mod2 = true
+                break
+            end
+        end
+        if !var_in_mod2
+            # We have found a predictor variable in model 1 that is not in model 2
+            return false 
+        end
+    end
+
+    return true
+end
+
+"""
+    `function ftest(mod::LinPredModel...)`
+Test if mod[i] fits significantly better than mod[i+1]
```

Comment:
No `function`, no backticks, line break after signature. Also I'd word this as, "For each pair of linear predictive models in `mod`, perform an F-test to determine whether one fits significantly better than the other."